src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,24 +1,159 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 /**
  * <p>Реалізувати свою HashMap, а саме методи `put(K key, V value)`, `getValue()` та `getSize()`.
  * Дотриматися основних вимог щодо реалізації мапи (initial capacity, load factor, resize...)
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double LOAD_FACTOR = 0.75;
+    private static final int CAPACITY = 16;
+    private int initialCapacity;
+    public Entry<K, V>[] table;
+    private int size = 0;
+
+    public MyHashMap() {
+        table = new Entry[CAPACITY];
+        initialCapacity = CAPACITY;
+    }
+
+    private int countBuckPos(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % table.length);
+    }
 
     @Override
     public void put(K key, V value) {
+        resise();
+        Entry<K, V> newEntry = new Entry<>(key, value);
+        int position = countBuckPos(newEntry.getKey());
+        Entry temp = table[position];
+        if (temp == null) {
+            table[position] = newEntry;
+            size++;
+            return;
+        }
 
+        while (temp != null) {
+            if (newEntry.getKey() == temp.getKey() || newEntry.getKey() != null
+                    && newEntry.getKey().equals(temp.getKey())) {
+                temp.setValue(value);
+                return;
+            }
+            if (temp.getNext() == null) {
+                temp.setNext(newEntry);
+                size++;
+                return;
+            }
+            temp = temp.getNext();
+        }
     }
 
     @Override
     public V getValue(K key) {
+        Entry temp;
+        if (key == null) {
+            temp = table[0];
+            if (temp.getKey() == null) {
+                return (V) temp.getValue();
```

Comment:
```suggestion
                return temp.getValue();
```
таке потрібно повидаляти скрізь